packages/solana/src/contracts/nft-drop.ts

```diff
@@ -0,0 +1,127 @@
+import { ClaimConditions } from "../classes/claim-conditions";
+import { NFTHelper } from "../classes/helpers/nft-helper";
+import { UserWallet } from "../classes/user-wallet";
+import { TransactionResult } from "../types/common";
+import { CommonNFTInput, NFTMetadata, NFTMetadataInput } from "../types/nft";
+import { Metaplex } from "@metaplex-foundation/js";
+import { Connection, PublicKey } from "@solana/web3.js";
+import { IStorage } from "@thirdweb-dev/storage";
+
+export class NFTDrop {
+  private connection: Connection;
+  private wallet: UserWallet;
+  private metaplex: Metaplex;
+  private storage: IStorage;
+  private nft: NFTHelper;
+  public dropMintAddress: PublicKey;
+  public claimConditions: ClaimConditions;
+
+  constructor(
+    dropMintAddress: string,
+    metaplex: Metaplex,
+    wallet: UserWallet,
+    storage: IStorage,
+  ) {
+    this.wallet = wallet;
+    this.storage = storage;
+    this.metaplex = metaplex;
+    this.connection = metaplex.connection;
+    this.nft = new NFTHelper(metaplex);
+    this.dropMintAddress = new PublicKey(dropMintAddress);
+    this.claimConditions = new ClaimConditions(dropMintAddress, metaplex);
+  }
+
+  async get(mintAddress: string): Promise<NFTMetadata> {
+    return this.nft.get(mintAddress);
+  }
+
+  async getAll(): Promise<NFTMetadata[]> {
+    const info = await this.getCandyMachine();
+    const nfts = await Promise.all(
+      info.items.map(async (item) => {
+        const metadata = await this.storage.get(item.uri);
+        return { uri: item.uri, ...metadata };
```

Comment:
Is there no address inside items we need to pass through here? Also name since it might not be in the JSON meta. 

Should work with candy machines not creates through us.

Reply:
Add a Todo to add a query count / start here like for EVM. Gotta be able to paginate this

Reply:
items includes unminted NFTs, so those don't have an address yet which is why we do it like this